pkg/controller/v1alpha2/applicationconfiguration/applicationconfiguration.go

```diff
@@ -347,6 +347,10 @@ func (r *OAMApplicationReconciler) updateStatus(ctx context.Context, ac *v1alpha
 	}
 	ac.Status.Workloads = append(ac.Status.Workloads, revisionStatus...)
 
+	if ac.Status.ObservedGeneration != ac.Generation {
+		ac.Status.ObservedGeneration = ac.Generation
```

Comment:
please make sure the ObservedGeneration is configured in every logic branch of the reconcile function.

Now it's only configured when the reconcile is succeed, but when one reconcile fail, users will also hope to see this CR is observed